components/script/task_source/mod.rs

```diff
@@ -8,9 +8,18 @@ pub mod history_traversal;
 pub mod networking;
 pub mod user_interaction;
 
+use dom::bindings::refcounted::Trusted;
+use dom::event::{EventBubbles, EventCancelable};
+use dom::eventtarget::EventTarget;
 use std::result::Result;
+use string_cache::Atom;
 
 pub trait TaskSource<T> {
     fn queue(&self, msg: T) -> Result<(), ()>;
+    fn queue_event(&self,
```

Comment:
Make this an inherent method, and pass `&EventTarget`, not `Trusted<EventTarget>`